src/mon/MonmapMonitor.h

```diff
@@ -29,16 +30,74 @@
 #include "MonMap.h"
 #include "MonitorDBStore.h"
 
+#include "command_multiplexer.h"
+
+class MMonGetMap;
 class MMonMap;
 class MMonCommand;
 class MMonJoin;
 
+namespace ceph::mon_cmds {
+
+class failure final 
```

Comment:
This should probably inherit from std::exception?

I guess it depends on how you feel about `catch (...)` versus `catch (const std::exception&)`

Reply:
Basically, I'm happy to make it derive from something-- perhaps std::runtime_error? But I'm equally (and very /slightly/ prefer) fine having it be an unrelated type, but you're definitely right in that it means having to use catch-all, which may not be desirable. I usually feel like there should be a layer of explicit catch and catch(...) used only at very specific points in a program, but YYMV as well!